### ISSUE TITLE
Add a new property type to manage a file.

### ISSUE THREAD

**Issue by antarus** [jhipster/jhipster-lite#4019]
Add a new property type to manage a file.

Let's imagine that my module requires a file to be configured.

**Comment by antarus** [comment]
I want to upload a dsl file for a new module, I would like to be able to add it in the frontend and retrieve it in the backend to process it

**Comment by antarus** [comment]
yes but at first, I may not need to interact with the other modules.

I was thinking more about the upload part of feature Replay #2907

But ideally your proposal seems the way to go for manage a dsl file.

**Comment by DamnClin** [comment]
The replay feature is not a module I think, for me it's a dedicated feature

**Comment by antarus** [comment]
I think adding a file type property to JHipsterPropertyType and managing it on the frontend side would do the job for some modules.

**Comment by DamnClin** [comment]
Yep but can't do it without a use case :D So we need a module that needs a file to add that

**Comment by pascalgrimaud** [comment]
agree with @DamnClin : it should be implemented only when you need it, not before. So there should be a use case for that

**Comment by antarus** [comment]
my module need a file in entry for generate code from this file. Il want to create my domain entity from it ans more later perhaps

**Comment by DamnClin** [comment]
I don't get it, by definition domain entity can't be generated since it's not related to any other kind of model (this is the main model, driving the other ones).

Also I don't think this is a module. Model generation (ex to generate a CRUD) is not a module, this is another feature, not a module

**Comment by DamnClin** [comment]
I can think of some modules that may need a file (modules related to a provider giving you a configuration file to add to your app) BUT adding file management is a lot of work (I think ~2h to handle the new type front + back and property type to get the file as a byte[]). I don't want to add it without any exact need
